src/publish/confluence/confluence-helper.ts

```diff
@@ -621,7 +629,58 @@ export const updateLinks = (
   const updatedChanges: ConfluenceSpaceChange[] =
     spaceChanges.map(changeMapper);
 
-  return updatedChanges;
+  return { pass1Changes: updatedChanges, pass2Changes: collectedPass2Changes };
+};
+
+export const convertForSecondPass = (
+  fileMetadataTable: Record<string, SitePage>,
+  spaceChanges: ConfluenceSpaceChange[],
+  server: string,
+  parent: ConfluenceParent
+): ConfluenceSpaceChange[] => {
+  const toUpdatesReducer = (
+    accumulator: ConfluenceSpaceChange[],
+    change: ConfluenceSpaceChange
+  ) => {
+    if (isContentUpdate(change)) {
+      accumulator = [...accumulator, change];
+    }
+
+    if (isContentCreate(change)) {
+      const qmdFileName = change?.fileName?.replace(".xml", ".qmd") ?? "";
```

Comment:
Just to play more file system defense here you could consider using base name to prevent a file name like `foo.xml.xml` from messing with you.

Reply:
Adding test, using `basename` now thanks again.